cli/compiler.rs

```diff
@@ -121,81 +190,50 @@ pub fn source_code_version_hash(
   gen_hash(vec![source_code, version.as_bytes(), config_hash])
 }
 
-fn load_config_file(
-  config_path: Option<String>,
-) -> (Option<PathBuf>, Option<Vec<u8>>) {
-  // take the passed flag and resolve the file name relative to the cwd
-  let config_file = match &config_path {
-    Some(config_file_name) => {
-      debug!("Compiler config file: {}", config_file_name);
-      let cwd = std::env::current_dir().unwrap();
-      Some(cwd.join(config_file_name))
-    }
-    _ => None,
-  };
-
-  // Convert the PathBuf to a canonicalized string.  This is needed by the
-  // compiler to properly deal with the configuration.
-  let config_path = match &config_file {
-    Some(config_file) => Some(config_file.canonicalize().unwrap().to_owned()),
-    _ => None,
-  };
-
-  // Load the contents of the configuration file
-  let config = match &config_file {
-    Some(config_file) => {
-      debug!("Attempt to load config: {}", config_file.to_str().unwrap());
-      match fs::read(&config_file) {
-        Ok(config_data) => Some(config_data.to_owned()),
-        _ => panic!(
-          "Error retrieving compiler config file at \"{}\"",
-          config_file.to_str().unwrap()
-        ),
-      }
-    }
-    _ => None,
-  };
-
-  (config_path, config)
-}
-
 pub struct TsCompiler {
-  pub deno_dir: DenoDir,
+  pub file_fetcher: SourceFileFetcher,
   pub config: CompilerConfig,
-  pub config_hash: Vec<u8>,
   pub disk_cache: DiskCache,
   /// Set of all URLs that have been compiled. This prevents double
   /// compilation of module.
   pub compiled: Mutex<HashSet<Url>>,
   /// This setting is controlled by `--reload` flag. Unless the flag
   /// is provided disk cache is used.
   pub use_disk_cache: bool,
+  /// This setting is controlled by `compilerOptions.checkJs`
+  pub compile_js: bool,
 }
 
 impl TsCompiler {
   pub fn new(
-    deno_dir: DenoDir,
+    file_fetcher: SourceFileFetcher,
+    disk_cache: DiskCache,
     use_disk_cache: bool,
     config_path: Option<String>,
-  ) -> Self {
-    let compiler_config = match load_config_file(config_path) {
-      (Some(config_path), Some(config)) => Some((config_path, config.to_vec())),
-      _ => None,
-    };
-
-    let config_bytes = match &compiler_config {
-      Some((_, config)) => config.clone(),
-      _ => b"".to_vec(),
+  ) -> Result<Self, ErrBox> {
+    let config = CompilerConfig::load(config_path)?;
+
+    // If `checkJs` is set to true in `compilerOptions` then we're gonna be compiling
+    // JavaScript files as well
+    let config_json = config.json()?;
+    let compile_js = match &config_json.get("compilerOptions") {
```

Comment:
Is the config parsed elsewhere in rust land ? If so maybe it makes sense to cache the parsed json object?

Reply:
No, that's the only place in Rust

Reply:
Just a health warning, `tsconfig.json` supports JSONC, which serde does not support.  This means that it will not be able to parse certain _valid_ `tsconfig.json` files.  When you scaffold a `tsconfig.json` via the `tsc` it includes comments, and so I think it will be a pretty common pattern in the wild to use JSONC.  So detecting `checkJs` may need to be done in another way, likely a regular expression of some sort.  It also might make sense to have the test fixtures have a `tsconfig.json` that includes comments.